File: app/src/main/java/com/example/soccernews/data/database/FavoriteDB.java
package com.example.soccernews.data.database;

import android.content.Context;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;

import com.example.soccernews.data.model.News;
import com.example.soccernews.data.services.FavoriteDao;

@Database(entities = {News.class}, version = 1)
public abstract class FavoriteDB extends RoomDatabase {

    public abstract FavoriteDao dao();
    static FavoriteDB INSTANCE = null;

    public static FavoriteDB getInstance(Context context) {

        if (INSTANCE == null) {
            FavoriteDB instance = Room.databaseBuilder(
                    context.getApplicationContext(),
                    FavoriteDB.class,
                    "favorite-news-db").build();
            INSTANCE = instance;
            return instance;
        } else {
            return INSTANCE;
        }

    }
}

File: app/src/main/java/viana/alison/soccernews/data/database/FavoriteDB.kt
package viana.alison.soccernews.data.database

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase
import viana.alison.soccernews.data.model.News
import viana.alison.soccernews.data.service.FavoriteDao

@Database(entities = [News::class], version = 1)
abstract class FavoriteDB : RoomDatabase() {

    abstract val favoriteDao: FavoriteDao

    companion object {

        @Volatile
        private var INSTANCE: FavoriteDB? = null

        fun getInstance(context: Context): FavoriteDB {
            return INSTANCE ?: synchronized(this) {
                val instance = Room.databaseBuilder(
                    context.applicationContext,
                    FavoriteDB::class.java,
                    "favorite-news-db")
                    .fallbackToDestructiveMigration()
                    .build()
                INSTANCE = instance
                instance
            }
        }

    }

}